src/OpenTracing/ISpan.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+
+namespace OpenTracing
+{
+    /// <summary>
+    /// Span represents an active, un-finished span in the OpenTracing system.
+    /// Spans are created by the <see cref="ITracer"/> interface.
+    /// </summary>
+    public interface ISpan : IDisposable
+    {
+        // TODO IDisposable behavior?
+        // TODO SetOperationName() ?
+        // TODO DateTimeOffset vs ticks ?
+
+        /// <summary>
+        /// Returns the <see cref="ISpanContext"/> for this Span. Note that the return
+        /// value of <see cref="Context"/> is still valid after a call to <see cref="Finish"/>, as is
+        /// a call to <see cref="Context"/> after a call to <see cref="Finish"/>.
+        /// </summary>
+        ISpanContext Context { get; } // TODO property or method?
+
+        /// <summary>
+        /// Adds a tag to the Span.
+        /// </summary>
+        /// <param name="key">If there is a pre-existing tag set for <paramref name="key"/>, it is overwritten.</param>
+        /// <param name="value">
+        /// Tag values can be numeric types, strings, or bools. The behavior of other tag value types is undefined 
+        /// at the OpenTracing level. If a tracing system does not know how to handle a particular value type, it
+        /// may ignore the tag, but shall not panic.
+        /// </param>
+        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
+        ISpan SetTag(string key, object value);
+
+        /// <summary>
+        /// Records an event with optional payload data for this Span.
+        /// </summary>
+        /// <param name="eventName">Name of the event.</param>
+        /// <param name="payload">An optional payload object.</param>
+        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
+        ISpan LogEvent(string eventName, object payload = null);
```

Comment:
just FYI, logging is a bit of a moving target right now... I'm going to make some progress on https://github.com/opentracing/opentracing.github.io/issues/96 today or this weekend.